### PR TITLE
#9054 case merge limit

### LINKED ISSUE #9054: Limit lists for duplicate merging of cases
### Problem Description

Data used for merging is currently only limited based on the report date filter, but still has to go through big parts of the available data.
See https://github.com/hzi-braunschweig/SORMAS-Project/issues/8697#issuecomment-1106285204

### Proposed Change

- [x] Limit `CaseFacadeEjb.getCasesForDuplicateMerging` method to fetch 100 entries.
- [x] ~~When 100 entries are returned display a warning below the grid in the UI: "The returned data has been limited to < placeholder > entries. You can adjust the creation date filter or add other filter criteria to drill down on this."~~
- [x] In addition show a drop-down that allows the user to define the limit: 100, 500, 1000, none
- [x] Re-evaluate the performance of this ~~with the database used in #8697~~. If the performance is still bad, provide an explain analyze via https://explain.dalibo.com/) If the performance is acceptable (max 10 seconds waiting time), the currently available warning popup shown when entering the merge view can be removed (including Strings.infoMergeFiltersHint)

### Implementation Details
- [x] get rid of joins that are not needed (see https://github.com/hzi-braunschweig/SORMAS-Project/issues/9054#issuecomment-1235243003)
* LEFT OUTER JOIN LOCATION location3_ ON person2_.address_id=location3_.id
* LEFT OUTER JOIN Region region4_ ON case0_.responsibleRegion_id=region4_.id
* LEFT OUTER JOIN Region region5_ ON case0_.region_id=region5_.id
* LEFT OUTER JOIN users user7_ ON case0_.reportingUser_id=user7_.id
* LEFT OUTER JOIN Facility facility8_ ON case0_.healthFacility_id=facility8_.id
* LEFT OUTER JOIN Region region10_ ON case1_.responsibleRegion_id=region10_.id
* LEFT OUTER JOIN Region region11_ ON case1_.region_id=region11_.id
- [x] get rid of distinct
- [x] introduce index to cases sort by creation date DESC

### Additional Information

- [x] Create issue to do the same for ContactFacadadeEjb.getContactsForDuplicateMerging #11469
- [x] Create issue to filter by archived/active and to only compare with cases that have creation date within or before date range. #11465
- [x] Create issue to provide guide on how to analyze & solve performance problems like these #11467

* After merging a duplicate, the list will be re-loaded, potentially causing a long delay. If we don't manage to improve the performance of the query itself we would at least have to optimize this aspect.

### PR DESCRIPTION
**Pull request by MartinWahnschaffe**
Fixes #9054

I have not adjusted the content of the existing warning in the duplicate view, because I have added a drop-down for the result count and was not sure how to adjust the existing text anyway.

I have tested it on a database with 85000 cases that result in 82000 potential duplicates, because they are generated. Using the limit of 100 or 1000 resulted in an execution of 1-3 seconds. Leaving the limitation away took more than 2 minutes.

### CODE REVIEW

**Comment by MartinWahnschaffe** [comment]
- [ ] provide explain analyze with timings for old query on big database
- [ ] force limitation to max 1000 duplicates

**Comment by MartinWahnschaffe** [comment]
Updated my "results" comment: https://github.com/hzi-braunschweig/SORMAS-Project/issues/9054#issuecomment-1420849461

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/QueryDetails.java [hunk 1]
```diff
@@ -6,13 +6,13 @@ public class QueryDetails extends BaseCriteria {
 
     public static final String RESULT_LIMIT = "resultLimit";
 
+    protected Integer resultLimit;
```
Comment (StefanKock) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/QueryDetails.java:9]: Make private, use getter & setter

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: Please find the results at https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11459

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: Please find the results at https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11459

**Pull request merged**